src/middleware.ts

```diff
@@ -0,0 +1,35 @@
+import { NextRequest, NextResponse } from 'next/server'
+import acceptLanguage from 'accept-language'
+import { fallbackLanguage, languages, cookieName } from './app/i18n/settings'
+
+acceptLanguage.languages(languages)
+
+export const config = {
+  // matcher: '/:language*'
+  matcher: ['/((?!api|_next/static|_next/image|assets|favicon.ico|sw.js).*)']
+}
+
+export function middleware(req: NextRequest) {
+  let language
+  if (req.cookies.has(cookieName)) language = acceptLanguage.get(req.cookies.get(cookieName)?.value)
```

Comment:
In cases like this it'd be best to also use curly braces in C-like languages. Makes for easier extensibility, but especially in JS you have [automatic semicolon insertion](http://en.wikipedia.org/wiki/JavaScript_syntax#Whitespace_and_semicolons) which can result in some very difficult to debug issues.

Reply:
For code I'd like to write it in an immutable way and use null-coalescing. 

```typescript
function getLanguageFromCookies(cookies: NextRequest['cookies']) {
  if (cookies.has(cookieName)) {
    return acceptLanguage.get(cookies.get(cookieName)?.value)
  }

  return undefined
}

function getLanguageFromHeaders(headers: Headers) {
  const language = headers.get('Accept-Language')
  return acceptLanguage.get(language)
}

export function middleware(req: NextRequest) {
  const language =
    getLanguageFromCookies(req.cookies) ??
    getLanguageFromHeaders(req.headers) ??
    fallbackLanguage
```

Here we also have the logic extracted to a simple to read function, so it's not that difficult to understand what the intent behind this code is.

Reply:
resolved